src/sentry/debug_files/artifact_bundles.py

```diff
@@ -204,3 +212,54 @@ def index_artifact_bundles_for_release(
         # there and wait for auto-expiration.
         metrics.incr("artifact_bundle_indexing.bundles_indexed")
         metrics.incr("artifact_bundle_indexing.urls_indexed", len(urls))
+
+
+def maybe_renew_artifact_bundles(used_artifact_bundles: Dict[int, datetime]):
+    if options.get("sourcemaps.artifact-bundles.enable-renewal") == 1.0:
+        with metrics.timer("artifact_bundle_renewal"):
+            renew_artifact_bundles(used_artifact_bundles)
+
+
+def renew_artifact_bundles(used_artifact_bundles: Dict[int, datetime]):
+    # We take a snapshot in time that MUST be consistent across all updates.
+    now = timezone.now()
+    # We compute the threshold used to determine whether we want to renew the specific bundle.
+    threshold_date = now - timedelta(days=AVAILABLE_FOR_RENEWAL_DAYS)
+
+    for (artifact_bundle_id, date_added) in used_artifact_bundles.items():
```

Comment:
I was thinking about a concern I had, what if we have here the same bundle_id? We are going to update the date_added in a non-deterministic order. Should we maybe de-dup the id and take the max date_added?